statsmodels/multivariate/cancorr.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+
+"""Canonical correlation analysis
+
+author: Yichuan Liu
+"""
+from __future__ import print_function, division
+
+from statsmodels.base.model import Model
+import numpy as np
+from numpy.linalg import svd
+from scipy import stats
+import pandas as pd
+from statsmodels.iolib import summary2
+from .multivariate_ols import multivariate_stats
+
+
+class CanCorr(Model):
+    """
+    Canonical correlation analysis using singluar value decomposition
+
+    For matrices x and y, find projections x_cancoef and y_cancoef such that:
+        x1 = x * x_cancoef, x1' * x1 is identity matrix
+        y1 = y * y_cancoef, y1' * y1 is identity matrix
+    and the correlation between x1 and y1 is maximized.
+
+    Attributes
+    -----------
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+    corr_values : array
+        The canonical correlation values
+    x_cancoeff: array
+        The canonical coefficients for exog
+    y_cancoeff: array
+        The canonical coeefficients for endog
+
+    .. [1] http://numerical.recipes/whp/notes/CanonCorrBySVD.pdf
+    .. [2] http://www.csun.edu/~ata20315/psy524/docs/Psy524%20Lecture%208%20CC.pdf
+    .. [3] http://www.mathematica-journal.com/2014/06/canonical-correlation-analysis/
+    """
+    def __init__(self, endog, exog, missing='none', hasconst=None, **kwargs):
+        super(CanCorr, self).__init__(endog, exog, **kwargs)
+
+    def fit(self, tolerance=1e-8):
+        """Fit the model
+
+        Parameters
+        ----------
+        tolerance : float
+            eigenvalue tolerance, values smaller than which is considered 0
+        Returns
+        -------
+
+        """
+        nobs, k_yvar = self.endog.shape
+        nobs, k_xvar = self.exog.shape
+        k = np.min([k_yvar, k_xvar])
+
+        x = np.array(self.exog)
+        x = x - x.mean(0)
+        y = np.array(self.endog)
+        y = y - y.mean(0)
+
+        ux, sx, vx = svd(x, 0)
+        # vx_ds = vx.T divided by sx
+        vx_ds = vx.T
+        mask = sx > tolerance
+        if mask.sum() < len(mask):
+            raise ValueError('exog is collinearity!')
```

Comment:
Just document it for now.
add to the docstring: 
"A ValueError is raised if there are singular values smaller than the tolerance. The treatment of singular arrays might change in future."